queequeg.js

```diff
@@ -32,113 +46,41 @@ function log_callback(type, request, response) {
-            'Content-Length': body.length
-        }
-    };
-
-    var post_req = http.request(post_options, function(response) {
-        var post_response = '';
-        response.setEncoding('utf8');
-        response.on('data', function(chunk) {
-            post_response = post_response + chunk;
-        });
-        response.on('end', function(chunk) {
-            post_response = post_response + (chunk || '');
+function addWebhook(hook, callback) {
+    console.log("== adding Webhook %s for request type %s", hookBaseUrl + "/" + hook, hook);
 
-            console.log("== reponse for request type %s: %s", hooks[0], post_response);
-            callback(hooks.slice(1));
-        });
+    singularity.webhooks.create(hook, hookBaseUrl, function(err, data) {
+        console.log("== reponse for request type %s: %s", hook, data);
+        callback(err);
     });
-
-    post_req.write(body);
-    post_req.end();
 }
 
-function delete_webhook(hooks, callback) {
-    var webhook = util.format('%s-http://%s:%s/webhook/%s', hooks[0].toUpperCase(), host, port, hooks[0])
-    console.log("== removing Webhook %s for request type %s", webhook, hooks[0]);
-
-    var delete_options =  {
-        host: singularity_host,
-        port : singularity_port,
-        path: singularity_base + 'api/webhooks/' + encodeURIComponent(webhook),
-        method: 'DELETE'
-    };
-
-    var delete_req = http.request(delete_options, function(response) {
-        var delete_response = '';
-        response.setEncoding('utf8');
-        response.on('data', function(chunk) {
-            delete_response = delete_response + chunk;
-        });
-        response.on('end', function(chunk) {
-            delete_response = delete_response + (chunk || '');
+function deleteWebhook(hook, callback) {
+    console.log("== removing Webhook %s for request type %s", hookBaseUrl + "/" + hook, hook);
 
-            console.log("== reponse for request type %s: %s", hooks[0], delete_response);
-            callback(hooks.slice(1));
-        });
+    singularity.webhooks.delete(hook, hookBaseUrl, function(err, data) {
+        console.log("== reponse for request type %s: %s", hook, data);
+        callback(null); // always return safely
     });
-
-    delete_req.end();
-}
-
-function add_webhooks(hooks) {
-    var add_callback = function(add_hooks) {
-        if (add_hooks.length > 0) {
-            add_webhook(add_hooks, add_callback);
-        }
-    };
-
-    add_webhook(hooks, add_callback);
-}
-
-function delete_webhooks(hooks, callback) {
-    var del_callback = function(del_hooks) {
-        if (del_hooks.length == 0) {
-            callback();
-        }
-        else {
-            delete_webhook(del_hooks, del_callback);
-        }
-    };
-
-    delete_webhook(hooks, del_callback);
 }
 
 var hooks = [ 'task', 'deploy', 'request' ];
 
 process.on('SIGINT', function() {
-    console.log("");
-    delete_webhooks(hooks,  function() { process.exit() });
+    console.log("CLEANUP");
```

Comment:
the reason for the newline was to have this message not show up in the middle of another.